src/cache.cpp

```diff
@@ -274,13 +275,16 @@ namespace {
 			return LoadDummyBitmap<T>(s.directory, f);
 		}
 
+#ifndef NDEBUG
 		// Test if the file was requested asynchronously before.
 		// If not the file can't be expected to exist -> bug.
+		// This text is expensive and turned off in release builds.
 		FileRequestAsync* request = AsyncHandler::RequestFile(s.directory, f);
 		if (!request->IsReady()) {
-			Output::Debug("BUG: File Not Requested: %s/%s", s.directory, f.c_str());
+			Output::Error("BUG: File Not Requested: %s/%s", s.directory, f.c_str());
```

Comment:
Error will currently close Player. Might be better to use Warning until #858.

Reply:
There are a lot of places where they just assume the cache will return non-null and then immediately start using the object. I'd rather we crash here with a big error message than get a mysterious segfault later on.

This error indicates a bug in Player code, and not game event code.

Open to suggestions though. What do you guys prefer?

Reply:
my original intention when writing this (with return nullptr) was to segfault right after to have a stacktrace.

Reply:
Agree let's use assert